core/error_macros.h

```diff
@@ -115,27 +115,40 @@ extern bool _err_error_exists;
 #endif
 
 #define ERR_FAIL_INDEX(m_index, m_size)                                                                                    \
-	do {                                                                                                                   \
+	{                                                                                                                      \
```

Comment:
As I mentioned before, the do {} while(0)  is a standard trick for the compiler to request you to not forget ; on that line, and also avoid warnings. As a more specific answer, yes, Qt creator shows the line with a yellow underline because there's an unnecesary ;